### PR TITLE
fix: a fresh clone runs (one config path, in-repo poster, owner from config, complete installer)

### PR DESCRIPTION
**Pull request by ThinkOffApp**
Closes items 1 to 4 of #23: after this, `git clone && ./install.sh && edit one file && systemctl enable --now` gives a cloner a car agent that starts, posts, and answers its own owner.

## What changes

1. **One config file for every module.** `carwatch/config.py` resolves it: `$CARWATCH_CONFIG`, then `$CARWATCH_STATE/config.json` (the systemd units already set `CARWATCH_STATE=~/.carwatch`), then `~/.carwatch/config.json`, then the legacy `/etc/carwatch/config.json`. agent, presence, listen, selfstate, obd_probe, mercedesme, webchat (four sites, including the wifi editor's write), `update.sh` and `switch-car.sh` all go through it. Shell helper: `python3 -m carwatch.config path|get KEY`.
2. **In-repo room poster.** `carwatch.room.post_as_car()` and `python3 -m carwatch.room --file F` replace `~/post-as-gle.py`, which only existed on the reference Pi. obdwatch, listen (both sites) and `pair-watch.sh` use it; `reach.sh` drops its dead poster variables. Never raises; a failed post is printed and the loop continues.
3. **Owner from config.** `config.json` `"owner"` drives the room gate (`agent._owner_ok`: the owner and `owner-*` devices; no owner configured = any human) and the presence dashboard target. Location facts no longer name a person or a desk (issue #15 class). Car identity defaults are neutral; the reference cars keep their descriptions through the `profiles/<handle>.json` overlay, unchanged.
4. **`install.sh` completes the install.** Seeds `~/.carwatch/config.json` (migrates an existing `/etc/carwatch/config.json`), generates the dash token, runs `apt-get update`, installs `alsa-utils bluez-alsa-utils ffmpeg network-manager`, and prints the whisper.cpp and piper steps the way it already did for the brain. `config.example.json` gains `owner` and a `car` block. `bench-results/` is gitignored (a voice recording sat there untracked).

## Tests

44 pass (`python3 -m unittest discover -s tests`), 10 new: config resolution order and CLI, owner gate, poster including `--file`. The two selfstate tests select the config through `CARWATCH_CONFIG` instead of patching `os.path.expanduser`.

## REVIEW_GATE

- [x] No hardcoded state the device should sense: this PR removes three (config path, owner, poster path) and adds none.
- [ ] **unverified-on-car.** Exercised on macOS against the unit tests only. Expected effect on the reference Pi: none (its config already lives at `~/.carwatch/config.json`, its handle overlay is intact), but that is a claim until `carwatch-update.timer` has run this on Vadelma and `journalctl -u carwatch-agent` shows the new `config ...; room gate: owner petrus` line.
- [x] Real deployment path: lands through `update.sh` like everything else; no manual copy.
- [x] Reachable the way the user reaches it: no network-path change.
- [x] Stopgaps labelled: the `/etc/carwatch/config.json` fallback is legacy support, not a second canonical location; `rfcomm.env` deliberately stays in `/etc/carwatch` because that unit runs as root.

Not in this PR (next, items 5 and 6 of #23): wiring the outbox into `post_as_car`, and the update.sh restart guard.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by ThinkOffApp — COMMENTED** [review]
Cross-review complete. The 44-test suite passes. I found one blocking issue before merge:

1. **P1 data loss: malformed/unreadable config can be overwritten.** `carwatch/config.py:61-70` makes `load_raw()` return `{}` for any open/JSON error. The new `/api/home-wifi/trust` handler then proceeds with that empty dict and calls `save_raw()` (`carwatch/webchat.py:3163-3173`), replacing the existing config with only `home_ssids`. A transient permission error or malformed/truncated config therefore erases `api_key`, `room`, `handle`, owner, and other settings when an authenticated dashboard request trusts/removes Wi-Fi. Please fail closed when the config cannot be loaded, and add a regression test that proves the original file remains unchanged.

2. **P2 precedence mismatch.** The docs say `$CARWATCH_CONFIG` wins always, but `config_path()` only selects it when the file already exists (`config.py:51-58`). If the explicit path is absent while `$CARWATCH_STATE/config.json` exists, reads and writes silently use the state file. Either make an explicit env path authoritative or document/test that it is only preferred when existing.

Also observed a small `ResourceWarning` from `carwatch/room.py:142` because the `--file` CLI leaves the body file open; not a merge blocker.

**Review by ThinkOffApp — COMMENTED** [review]
Cross-review completed. I found one actionable issue in the new wifi trust path: `load_raw()` intentionally returns `{}` on malformed or unreadable config, but `/api/home-wifi/trust` then passes that empty dict to `save_raw()`. A transient permission error or partial write can therefore replace the entire config with only `home_ssids`, erasing `api_key`, `room`, and the car identity. Please fail closed when the resolved config cannot be loaded (or validate required keys before saving), and add a regression test for unreadable/invalid config. The config resolver order and normal `save_raw` path otherwise look coherent; the 44-test suite passes.

**Review by ThinkOffApp — COMMENTED** [review]
Cross-review complete. The config-resolution order itself looks correct, and the branch's claimed suite passes: 44 tests OK. One blocking data-loss edge case remains in the Wi-Fi editor: `load_raw()` intentionally converts malformed or unreadable JSON into `{}`, but this handler immediately passes that result to `save_raw()` and replaces the existing config with only `home_ssids`. A transient partial write, invalid manual edit, or permission/read error can therefore erase the car's api_key, room, handle, owner, and all other settings when Petrus taps trust/remove. Please make this write path fail closed on parse/read errors and add a regression test for malformed-config preservation.

File: carwatch/webchat.py [hunk 1]
```diff
@@ -3161,20 +3160,17 @@ def val(group, key):
...
                         "application/json")
-                cfg_path = os.path.expanduser("~/.carwatch/config.json")
-                with open(cfg_path) as fh:
-                    cfg = json.load(fh)
+                from carwatch.config import config_path, load_raw, save_raw
+                cfg_path = config_path()
+                cfg = load_raw(cfg_path)
                 homes = [s for s in (cfg.get("home_ssids") or []) if s]
                 remove = bool(body.get("remove"))
                 if remove:
```
Comment (ThinkOffApp) [inline carwatch/webchat.py]: [P1] Do not overwrite an unreadable config. `load_raw()` returns `{}` for malformed or unreadable JSON, after which `save_raw()` replaces the whole config with only `home_ssids`, potentially deleting api_key, room, owner, and every other setting. This endpoint should fail closed on parse/read errors and have a regression test.

**Comment by ThinkOffApp** [comment]
All four findings addressed in 49c626a, thank you both.

- **P1 config clobber (codexmb):** the wifi trust editor now uses `config.update_config()`, a strict read-modify-write that raises and writes nothing when the file is missing, unreadable or malformed. Regression test `test_update_config_fails_closed_on_malformed_file` writes a truncated config, attempts the edit, and asserts the file is byte-identical afterwards; the happy path asserts every other key survives.
- **P2 precedence (codexmb + codex):** an explicit `$CARWATCH_CONFIG` is now authoritative even when the file does not exist, so a typo fails visibly. Test `test_explicit_env_wins_even_when_missing`.
- **P1 voice notes (codex):** room voice notes pass `_owner_ok(sender, owner)`, the same gate as text.
- **P2 identity (codex):** `car_identity()` layers neutral defaults < repo profile < config `car` block, so a config with a handle and no car block keeps its GLE identity. Test `test_profile_overlays_neutral_defaults` covers all three layers.
- `room` CLI `--file` closes its handle; suite runs clean with `-W error::ResourceWarning`.

47 tests pass. Still labelled unverified-on-car.

**Review by ThinkOffApp — COMMENTED** [review]
Re-review of commit 49c626a222dd900667f2fa42ed992671f7331058: all previously reported findings are addressed. The strict config read-modify-write fails closed and has preservation coverage; explicit CARWATCH_CONFIG precedence has coverage; voice notes use the owner gate; profile layering and the --file ResourceWarning fix are covered. Independently ran the full suite with ResourceWarning promoted to errors: 47 tests passed. No further blocking or non-blocking findings.

**Comment by ThinkOffApp** [comment]
Follow-up only (non-blocking): `post_as_car(timeout=...)` accepts a timeout argument but `RoomClient.post()` still hardcodes `timeout=20`. Consider wiring the argument through or removing the unused parameter in a later cleanup. The requested four fixes remain verified and the PR is merge-ready, subject to the documented on-car verification.
